worlds/yugioh06/rules.py

```diff
@@ -0,0 +1,878 @@
+        "Revival Jam",
+        "Yomi Ship",
+        "Treeborn Frog"
+    ], player) >= 2 and state.has("Mother Grizzly", player)
+
+
+def only_fire(state, player):
+    return state.count_from_list_exclusive([
+        "Blazing Inpachi",
+        "Familiar-Possessed - Hiita",
+        "Great Angus",
+        "Fire Beaters"
+    ], player) >= 2 and state.count_from_list_exclusive([
+        "Thestalos the Firestorm Monarch",
+        "Horus the Black Flame Dragon LV6"
+    ], player) >= 1 and state.count_from_list_exclusive([
+        "Solar Flare Dragon",
+        "Tenkabito Shien",
+        "Ultimate Baseball Kid"
+    ], player) >= 2 and state.has("UFO Turtle", player)
+
+
+def only_wind(state, player):
+    return state.count_from_list_exclusive([
+        "Luster Dragon",
+        "Slate Warrior",
+        "Spear Dragon",
+        "Familiar-Possessed - Wynn",
+        "Harpie's Brother",
+        "Nin-Ken Dog",
+        "Cyber Harpie Lady",
+        "Oxygeddon"
+    ], player) >= 2 and state.count_from_list_exclusive([
+        "Cyber-Tech Alligator",
+        "Luster Dragon #2",
+        "Armed Dragon LV5",
+        "Roc from the Valley of Haze"
+    ], player) >= 1 and state.count_from_list_exclusive([
+        "Armed Dragon LV3",
+        "Twin-Headed Behemoth",
+        "Harpie Lady 1"
+    ], player) >= 2 and state.has("Flying Kamakiri 1", player)
+
+
+def only_fairy(state, player):
+    return state.count_from_list_exclusive([
+        "Dunames Dark Witch",
+        "Hysteric Fairy"
+    ], player) >= 1 and (state.count_from_list_exclusive([
+        "Dunames Dark Witch",
+        "Hysteric Fairy",
+        "Dancing Fairy",
+        "Zolga",
+        "Shining Angel",
+        "Kelbek",
+        "Mudora",
+        "Asura Priest",
+        "Cestus of Dagla"
+    ], player) + (state.count_from_list_exclusive([
+        "The Agent of Judgment - Saturn",
+        "Airknight Parshath"
+    ], player) >= 1)) >= 7
+
+
+def only_warrior(state, player):
+    return state.count_from_list_exclusive([
+        "Dark Blade",
+        "Blindly Loyal Goblin",
+        "D.D. Survivor",
+        "Gearfried the Iron knight",
+        "Ninja Grandmaster Sasuke",
+        "Warrior Beaters"
+    ], player) >= 1 and (state.count_from_list_exclusive([
+        "Dark Blade",
+        "Blindly Loyal Goblin",
+        "D.D. Survivor",
+        "Gearfried the Iron Knight",
+        "Ninja Grandmaster Sasuke",
+        "Warrior Beaters",
+        "Warrior Lady of the Wasteland",
+        "Exiled Force",
+        "Mystic Swordsman LV2",
+        "Dimensional Warrior",
+        "Dandylion",
+        "D.D. Assailant",
+        "Blade Knight",
+        "D.D. Warrior Lady",
+        "Marauding Captain",
+        "Command Knight",
+        "Reinforcement of the Army"
+    ], player) + (state.count_from_list_exclusive([
+        "Freed the Matchless General",
+        "Holy Knight Ishzark",
+        "Silent Swordsman Lv5"
+    ], player) >= 1)) >= 7
```

Comment:
For the record, this is a place where `state.count_[...]` is warranted because you are then further manipulating the count / adding to it.

*Comment*